cloudigrade/api/tests/clouds/azure/util/test_create_initial_azure_instance_events.py

```diff
@@ -36,8 +36,8 @@ def test_create_initial_azure_instance_events(self, mock_enable):
         self.assertTrue(AzureInstance.objects.count() == 2)
         self.assertTrue(AzureInstance.objects.filter(resource_id=vm1["id"]).exists())
         self.assertTrue(AzureInstance.objects.filter(resource_id=vm2["id"]).exists())
-        self.assertTrue(AzureInstanceEvent.objects.count() == 1)
-        self.assertTrue(InstanceEvent.objects.count() == 1)
+        self.assertTrue(AzureInstanceEvent.objects.count() == 2)
+        self.assertTrue(InstanceEvent.objects.count() == 2)
         self.assertEqual(
-            InstanceEvent.objects.first().event_type, InstanceEvent.TYPE.power_on
+            InstanceEvent.objects.first().event_type, InstanceEvent.TYPE.power_off
```

Comment:
What caused this to change? If we were asserting something about `power_on` is created before this change, should there be another test or more assertions to ensure the `power_on` is still created?

Reply:
It's because how the test is set up originally, it's looking for one InstanceEvent with the state `power_on`, however, now that we add an InstanceEvent regardless of state, the first instance event happens to be for vm1 which is not running and in the `power_off` state. I can add another line that asserts that the last object is `power_on` (ie show that both states are covered in InstanceEvents.

Reply:
So basically I can add another line. The first one corresponds to vm1 which is not running (thus is in a `power_off` state), and vm2 with the `power_on` state. Or I can do a count, and that there's one `power_off` InstanceEvent and one `power_on` InstanceEvent :

```python
    self.assertEqual(
        InstanceEvent.objects.first().event_type, InstanceEvent.TYPE.power_off
    )
    self.assertEqual(
        InstanceEvent.objects.last().event_type, InstanceEvent.TYPE.power_on
    )
```

Reply:
Yes, let's add the additional assertion to find the `power_on` event since confirming its existence was probably the original intention of that test.

Reply:
Awesome, I've added that line to the test